tests/cache_pools_test.py

```diff
@@ -63,29 +63,53 @@ def test_sharded_cache_pool_different_order_same_results(mocker: MockerFixture)
     assert connection_pool._server == "2.2.2.2:11211"
 
 
+def test_sharded_cache_pool_honors_server_id(mocker: MockerFixture) -> None:
+    mocker.patch("meta_memcache.configuration.socket", autospec=True)
+    server_addresses = [
+        ServerAddress(host="1.1.1.1", port=11211, server_id="1"),
+        ServerAddress(host="2.2.2.2", port=11211, server_id="2"),
+    ]
+    cache_pool_a = ShardedCachePool.from_server_addresses(
+        servers=server_addresses,
+        connection_pool_factory_fn=connection_pool_factory_builder(),
+    )
+    server_addresses = [
+        ServerAddress(host="1.1.1.1", port=11211, server_id="2"),
+        ServerAddress(host="2.2.2.2", port=11211, server_id="1"),
+    ]
+    cache_pool_b = ShardedCachePool.from_server_addresses(
+        servers=server_addresses,
+        connection_pool_factory_fn=connection_pool_factory_builder(),
+    )
+    connection_pool = cache_pool_a._get_pool(Key("foo"))
+    assert connection_pool._server == "1"
```

Comment:
Why use `_server`? SHouldn't be that a private field?

Reply:
This is a test, for ensuring correct behavior it is legit for a test to use private fields...

Reply:
Yeah, my question is if we need it to stay private or we can expose it

Reply:
Oh, sure, I guess we can expose it.